subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -530,19 +541,76 @@ static int cell_pos_cb_send(const char *const rx_buf)
 	return -EFTYPE;
 }
 
+void agps_process(const char * const buf, const size_t buf_len)
+{
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+	int ret = nrf_cloud_agps_process(buf, buf_len);
+
+	LOG_DBG("A-GPS data processed; result: %d", ret);
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+	if (ret == 0) {
+		ret = nrf_cloud_pgps_notify_prediction();
+		if (ret) {
+			LOG_ERR("Error requesting P-GPS notification: %d", ret);
+		}
+	}
+#endif
+#endif
+}
+
+void pgps_process(const char * const buf, const size_t buf_len)
+{
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+	int ret = nrf_cloud_pgps_process(buf, buf_len);
+
+	LOG_DBG("P-GPS data processed; result: %d", ret);
+#endif
+}
+
+int cell_pos_process(const char * const buf)
+{
+#if defined(CONFIG_NRF_CLOUD_CELL_POS)
+	if (cell_pos_cb_send(buf) == 0) {
+		LOG_DBG("Cellular positioning data sent to provided callback");
+		return 0;
+	}
+#endif
+	return -ENOSYS;
+}
+
 static int dc_rx_data_handler(const struct nct_evt *nct_evt)
 {
+	bool discon_req = false;
+
 	struct nrf_cloud_evt cloud_evt = {
-		.type = NRF_CLOUD_EVT_RX_DATA,
 		.data = nct_evt->param.dc->data,
 		.topic = nct_evt->param.dc->topic,
 	};
 
```

Comment:
It would be nice to have the topic logged here, ideally with the data, though that's problematic for A-GPS.